core/services/relay/evm/chain_writer.go

```diff
@@ -0,0 +1,82 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/google/uuid"
+
+	commonservices "github.com/smartcontractkit/chainlink-common/pkg/services"
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
```

Comment:
nit since you'll need to push a commit to fix the merge conflict anyways: Think we should group this import with the `commonservices` one above since both are from `chainlink-common`

Reply:
Actually I forgot you'd maintain the approvals after resolving the merge conflicts so you can ignore this. Not important enough to try to get approvals again